classes/class-woothemes-sensei-certificates.php

```diff
@@ -878,63 +928,30 @@ public function certificate_text( $pdf_certificate, $fpdf ) {
 
-				$certificate_message = $this->certificate_template_fields['certificate_message']['text'];
-				$certificate_message = str_replace( array( '{{learner}}', '{{course_title}}', '{{completion_date}}', '{{course_place}}' ), array( $student_name, $course_title, $date, get_bloginfo( 'name' ) ), $certificate_message );
-
-			} // End If Statement
-
-			$certificate_course = $course_title;
-			if ( isset( $this->certificate_template_fields['certificate_course']['text'] ) && '' != $this->certificate_template_fields['certificate_course']['text'] ) {
-
-				$certificate_course = $this->certificate_template_fields['certificate_course']['text'];
-				$certificate_course = str_replace( array( '{{learner}}', '{{course_title}}', '{{completion_date}}', '{{course_place}}' ), array( $student_name, $course_title, $date, get_bloginfo( 'name' ) ), $certificate_course );
-
-			} // End If Statement
+				$meta_key = 'certificate_' . $field_key;
 
-			$certificate_completion = $date; // {{completion_date}}
-			if ( isset( $this->certificate_template_fields['certificate_completion']['text'] ) && '' != $this->certificate_template_fields['certificate_completion']['text'] ) {
-
-				$certificate_completion = $this->certificate_template_fields['certificate_completion']['text'];
-				$certificate_completion = str_replace( array( '{{learner}}', '{{course_title}}', '{{completion_date}}', '{{course_place}}' ), array( $student_name, $course_title, $date, get_bloginfo( 'name' ) ), $certificate_completion );
-
-			} // End If Statement
-
-			$certificate_place = sprintf( __( 'At %s', 'sensei-certificates' ), get_bloginfo( 'name' ) ); // At {{course_place}}
-			if ( isset( $this->certificate_template_fields['certificate_place']['text'] ) && '' != $this->certificate_template_fields['certificate_place']['text'] ) {
-
-				$certificate_place = $this->certificate_template_fields['certificate_place']['text'];
-				$certificate_place = str_replace( array( '{{learner}}', '{{course_title}}', '{{completion_date}}', '{{course_place}}' ), array( $student_name, $course_title, $date, get_bloginfo( 'name' ) ), $certificate_place );
-
-			} // End If Statement
-
-			$output_fields = array(
-				'certificate_heading'    => 'text_field',
-				'certificate_message'    => 'textarea_field',
-				'certificate_course'     => 'text_field',
-				'certificate_completion' => 'text_field',
-				'certificate_place'      => 'text_field',
-			);
+				// Get the default field value.
+				$field_value = $field_info['text_placeholder'];
+				if ( isset( $this->certificate_template_fields[ $meta_key ]['text'] ) && '' !== $this->certificate_template_fields[ $meta_key ]['text'] ) {
+					$field_value = $this->certificate_template_fields[ $meta_key ]['text'];
+				}
 
-			foreach ( $output_fields as $meta_key => $function_name ) {
+				// Replace the template tags.
+				$field_value = apply_filters( 'sensei_certificate_data_field_value', $field_value, $field_key, $student, $course );
```

Comment:
Maybe add a doc block for the filter?
I know we don't build docs for hooks here, but it just feels right 😅

Reply:
Great suggestion, thanks! I've updated the docs for both new filters in https://github.com/woocommerce/sensei-certificates/pull/292/commits/f9b2e2e835d7b96f92dc8f696f623f18a8301e41.